cmake/PureCMakeBuild.cmake

```diff
@@ -0,0 +1,290 @@
+cmake_minimum_required(VERSION 2.6)
+
+if (NOT CMAKE_VERSION VERSION_LESS "3.1")
+	cmake_policy(SET CMP0054 NEW)
+endif ()
+
+set(LIB_NAME nabo)
+project("lib${LIB_NAME}")
+
+# Extract version from header
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR})
+execute_process(
+	COMMAND grep "NABO_VERSION " nabo/nabo.h
+	WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+	RESULT_VARIABLE GREP_VERSION_RESULT
+	OUTPUT_VARIABLE PROJECT_VERSION
+	OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+if (NOT GREP_VERSION_RESULT EQUAL 0)
+	message(SEND_ERROR "Cannot grep version number: ${GREP_VERSION_RESULT}")
+endif ()
+string(REGEX REPLACE ".*\"(.*)\".*" "\\1" PROJECT_VERSION "${PROJECT_VERSION}" )
+
+if (NOT CMAKE_BUILD_TYPE)
+	message("-- No build type specified; defaulting to CMAKE_BUILD_TYPE=Release.")
+	set(CMAKE_BUILD_TYPE Release CACHE STRING
+	  "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel."
+	  FORCE)
+else ()
+	if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+		message("\n=================================================================================")
+		message("\n-- Build type: Debug. Performance will be terrible!")
+		message("-- Add -DCMAKE_BUILD_TYPE=Release to the CMake command line to get an optimized build.")
+		message("\n=================================================================================")
+	endif ()
+endif ()
+
+if (NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
+	add_definitions(-O3)
+endif()
+
+# Documentation
+option(LIBNABO_BUILD_DOXYGEN "Build libnabo doxygen documentation" ON)
+if (LIBNABO_BUILD_DOXYGEN)
```

Comment:
Why isn't this in the common part for both pure cmake and catkin?

Reply:
You're right, Doxygen is processed "automagically" for ROS packages by the build farm is you enable the doc job.